api_app/api/routes/airlock_resource_helpers.py

```diff
@@ -63,5 +79,75 @@ async def save_airlock_review(airlock_review: AirlockReview, airlock_review_repo
+                                         constants.CORE_RESOURCE_GROUP_NAME.format(tre_id))
+        elif airlock_request.status == AirlockRequestStatus.Submitted:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_IMPORT_INPROGRESS.format(tre_id),
+                                         constants.CORE_RESOURCE_GROUP_NAME.format(tre_id))
+        elif airlock_request.status == AirlockRequestStatus.InReview:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_IMPORT_INPROGRESS.format(tre_id),
+                                         constants.CORE_RESOURCE_GROUP_NAME.format(tre_id))
+        elif airlock_request.status == AirlockRequestStatus.Approved:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_IMPORT_APPROVED.format(short_workspace_id),
+                                         constants.WORKSPACE_RESOURCE_GROUP_NAME.format(tre_id, short_workspace_id))
+        elif airlock_request.status == AirlockRequestStatus.Rejected:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_IMPORT_REJECTED.format(tre_id),
+                                         constants.CORE_RESOURCE_GROUP_NAME.format(tre_id))
+    else:
+        if airlock_request.status == AirlockRequestStatus.Draft:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_EXPORT_INTERNAL.format(short_workspace_id),
+                                         constants.WORKSPACE_RESOURCE_GROUP_NAME.format(tre_id, short_workspace_id))
+        elif airlock_request.status in AirlockRequestStatus.Submitted:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_EXPORT_INPROGRESS.format(short_workspace_id),
+                                         constants.WORKSPACE_RESOURCE_GROUP_NAME.format(tre_id, short_workspace_id))
+        elif airlock_request.status == AirlockRequestStatus.InReview:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_EXPORT_INPROGRESS.format(short_workspace_id),
+                                         constants.WORKSPACE_RESOURCE_GROUP_NAME.format(tre_id, short_workspace_id))
+        elif airlock_request.status == AirlockRequestStatus.Approved:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_EXPORT_APPROVED.format(tre_id),
+                                         constants.CORE_RESOURCE_GROUP_NAME.format(tre_id))
+        elif airlock_request.status == AirlockRequestStatus.Rejected:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_EXPORT_REJECTED.format(short_workspace_id),
+                                         constants.WORKSPACE_RESOURCE_GROUP_NAME.format(tre_id, short_workspace_id))
+
+
+def validate_user_is_allowed_to_access_sa(user: User, airlock_request: AirlockRequest):
+    if "WorkspaceResearcher" not in user.roles and airlock_request.status != AirlockRequestStatus.InReview:
```

Comment:
We don't, this how it is done in all the other places, which is all the auth, and I prefer not to touch those areas in this PR, if you'd like I can create a separate story to do it, because I agree we should